src/pass/pass.service.ts

```diff
@@ -1,26 +1,87 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
 import { CreatePassDto } from './dto/create-pass.dto';
 import { UpdatePassDto } from './dto/update-pass.dto';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { PassInterface } from './interfaces/pass.interface';
+import { UserInterface } from '../user/interfaces/user.interface';
 
 @Injectable()
 export class PassService {
-  create(createPassDto: CreatePassDto) {
-    return 'This action adds a new pass';
+  constructor(
+    @InjectModel('Pass') private readonly passModel: Model<PassInterface>,
+    @InjectModel('User') private readonly userModel: Model<UserInterface>, //@InjectModel('Group') private  readonly groupModel: Model<GroupInterface>
+  ) {}
+
+  async findAll(): Promise<PassInterface[]> {
+    const passes = await this.passModel.find().populate('user', '-password -gender').exec();
+
+    if (!passes || !passes[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return passes;
+  }
+
+  async findByID(id: string): Promise<PassInterface> {
+    const pass = await this.passModel.findOne({ _id: id }).populate('user', '-password -gender').exec();
+    if (!pass) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return pass;
   }
 
-  findAll() {
-    return `This action returns all pass`;
+  async findAllByUserID(userId: string): Promise<PassInterface[]> {
+    const passes = await this.passModel.find().populate('user', '-password -gender').exec();
+    let filtered = passes.filter((pass) => {
+      return pass.user._id == userId;
+    });
+    if (!filtered || !filtered[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return filtered;
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} pass`;
+  /*async findAllByGroupID(groupId: string): Promise<PassInterface[]> {
+    const passes = await this.passModel
+    .find()
+    .populate('group')
+    .exec()
+    let filtered = passes.filter((pass) => {
+      return pass.group == groupId;
+    })
+    if (!filtered || !filtered[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return filtered
+  }*/
+
+  async create(createPassDto: CreatePassDto) {
+    const user = await this.userModel.findById(createPassDto.user).exec();
+
+    /*const group = await this.groupModel.findById(createPassDto.group)
+        .exec();*/
+
+    if (user) {
+      const pass = await new this.passModel(createPassDto);
+      return pass.save();
+    } else {
+      throw new HttpException('User or Group Not Found', HttpStatus.NOT_FOUND);
+    }
```

Comment:
we wszystkich pozostałych metodach sprawdzasz najpierw czy dany element nie istnieje czyli if(!user), tutaj jako w jedynej na odwrót ;D można zostawić, ale interesujące :D

Reply:
no bo tutaj nie szukam pass tylko user. Jak tej walidacji nie było to można by było stworzyć pass dla nie istniejącego usera -,nawet jest napisany na to test 'should not create pass to not existing user'

Reply:
taaak, ja wiem, chodzi mi tylko ze tutaj jest if (exist)(do sth) else (error), a we wszystkich innych masz if(not exist) throw error, a ponizej reszta, sam sposob zapisu :D  a dokładniej to 
`if(!user) throw new HttpException('User or Group Not Found', HttpStatus.NOT_FOUND);
const pass = await new this.passModel(createPassDto);
return pass.save();`
tak mi się w oczy rzuciło tylko, że w pozostałych masz inaczej, ale tak też jest jak najbardziej git. A walidacja oczywiście, że dobra!